graylog2-web-interface/src/components/common/FormSubmit.tsx

```diff
@@ -24,60 +24,71 @@ import Spinner from 'components/common/Spinner';
 
 type WithCancelProps = {
   displayCancel: true,
-  disableCancel?: boolean,
+  disabledCancel?: boolean,
   onCancel: () => void,
 }
 
 type WithoutCancelProps = {
   displayCancel: false
 }
 
+type WithAsyncSubmit = {
+  isAsyncSubmit: true,
+  submitLoadingText: string,
+  isSubmitting: boolean,
+}
+
+type WithSyncSubmit = {
+  isAsyncSubmit: false,
+}
+
 type Props = {
   bsSize?: 'large' | 'small' | 'xsmall',
+  centerCol?: React.ReactNode,
   className?: string,
-  displayCancel?: boolean,
   disabledSubmit?: boolean,
+  displayCancel?: boolean,
   formId?: string,
-  isSubmitting?: boolean,
+  isAsyncSubmit?: boolean,
   onSubmit?: () => void,
   submitButtonText: string,
-  submitIcon?: IconName,
   submitButtonType?: 'submit' | 'button',
-  submitLoadingText?: string,
-} & (WithCancelProps | WithoutCancelProps)
+  submitIcon?: IconName,
+} & (WithCancelProps | WithoutCancelProps) & (WithAsyncSubmit | WithSyncSubmit);
 
 const FormSubmit = (props: Props) => {
   const {
     bsSize,
     className,
+    centerCol,
     displayCancel,
     disabledSubmit,
     formId,
-    isSubmitting,
+    isAsyncSubmit,
     onSubmit,
     submitButtonText,
     submitButtonType,
     submitIcon,
-    submitLoadingText,
   } = props;
 
   return (
     <ButtonToolbar className={className}>
       <Button bsStyle="success"
               bsSize={bsSize}
-              disabled={disabledSubmit || isSubmitting}
+              disabled={disabledSubmit || (isAsyncSubmit && props.isSubmitting)}
               form={formId}
               title={submitButtonText}
               type={submitButtonType}
               onClick={onSubmit}>
-        {(submitIcon && !isSubmitting) && <><Icon name={submitIcon} /> </>}
-        {isSubmitting ? <Spinner text={submitLoadingText} delay={0} /> : submitButtonText}
+        {(submitIcon && !(isAsyncSubmit && props.isSubmitting)) && <><Icon name={submitIcon} /> </>}
```

Comment:
I think we can remove Fragment here `<><Icon name={submitIcon} /> </> ` or you need space?